jobs/buildhub/check_missing_releases.py

```diff
@@ -0,0 +1,335 @@
+    version_url = archive_url(product, version)
+    platform_folders, _ = await fetch_listing(loop, session, version_url)
+
+    platforms = [p[:-1] for p in platform_folders]  # strip trailing /
+    platforms = [p for p in platforms if p not in ("source", "update", "contrib")]
+
+    futures = [fetch_locales(loop, session, queue, product, version, platform)
+               for platform in platforms]
+    return await asyncio.gather(*futures)
+
+
+async def fetch_locales(loop, session, queue, product, version, platform):
+    platform_url = archive_url(product, version, platform)
+    locale_folders, _ = await fetch_listing(loop, session, platform_url)
+
+    locales = [l[:-1] for l in locale_folders]
+    locales = [l for l in locales if l != "xpi"]
+
+    futures = [fetch_files(loop, session, queue, product, version, platform, locale)
+               for locale in locales]
+    return await asyncio.gather(*futures)
+
+
+async def fetch_files(loop, session, queue, product, version, platform, locale):
+    locale_url = archive_url(product, version, platform, locale)
+    _, files = await fetch_listing(loop, session, locale_url)
+
+    futures = []
+    for file_ in files:
+        filename = file_["name"]
+        url = locale_url + filename
+
+        if not is_release_filename(product, filename):
+            print('Skipping', url)
+            continue
+
+        size = file_["size"]
+        date = file_["last_modified"]
+
+        metadata = await fetch_release_metadata(session, product, version, platform, locale)
+        record = archive(product, version, platform, locale, url, size, date, metadata)
+        logger.debug("Release found {}".format(url))
+
+        futures.append(queue.put(record))
+
+    return await asyncio.gather(*futures)
+
+
+async def fetch_products(loop, session, queue, products, client):
+    # Nightlies
+    # futures = [fetch_nightlies(loop, session, queue, product, client) for product in products]
+    # await asyncio.gather(*futures)
+    # Releases
+    futures = [fetch_versions(loop, session, queue, product, client) for product in products]
+    await asyncio.gather(*futures)
+
+
+async def produce(loop, queue, client):
+    """Grab releases from the archives website."""
+    async with aiohttp.ClientSession(loop=loop) as session:
+        await fetch_products(loop, session, queue, PRODUCTS, client)
+    logger.info("Scraping releases done.")
+
+
+async def consume(loop, queue, executor, client):
+    """Store grabbed releases from the archives website in Kinto."""
+
+    # Check for 200 or 404 on the release name
+    print("Start consumer.")
+
+    while True:
+        record = await queue.get()
+        release_url = record['download']['url']
+        release_id = build_record_id(release_url)
+        if release_id is None:
+            print('Skipping', release_url)
+            continue
+        try:
+            client.get_record(id=release_id)
```

Comment:
It was not supposed to find lots of them. I guess the best way is to create a worker to would consume a list of archive that needs to be added to Kinto and other scrpts that would populate that list. So that we can have different kind of way to populate that list (S3 for instance).

The plan of this script was to validate the work done in `scraping_archive.py` so merging them defeat the purpose of this script I guess.